src/isolate/server/_auth.py

```diff
@@ -0,0 +1,16 @@
+import jwt
+
+ALGORITHM = 'HS256'
+
+def create_auth_token(user_key: str, secret_key: str) -> str:
+    # Encode user_key in a JWT
+    return jwt.encode({'key': user_key}, secret_key, ALGORITHM)
+
+def validate_auth_token(token: str, user_key: str, secret_key: str) -> bool:
+    # Decode user_key from JWT
+    try:
+        decoded = jwt.decode(token, secret_key, algorithms=[ALGORITHM])
+        decoded_key = decoded['key']
+        return user_key == decoded_key
+    except Exception:
+        return False
```

Comment:
I think we can limit the exception handling a bit to only what `jwt.decode` can raise.
```suggestion
    try:
        decoded = jwt.decode(token, secret_key, algorithms=[ALGORITHM])
    except jwt.PyJWTError:
        return False
    else:
        return user_key == decoded.get('key')
```